src/bspatch.rs

```diff
@@ -169,22 +170,15 @@ where
         bsize: usize,
         dsize: usize,
     ) -> Self {
-        let mut buf = Vec::with_capacity(bsize);
-        let mut dlt = Vec::with_capacity(dsize);
-        unsafe {
-            buf.set_len(bsize);
-            dlt.set_len(dsize);
-        }
-
         Context {
             source: Cursor::new(source),
             target,
             ctrls,
             delta,
             extra,
             n: 0,
-            buf,
-            dlt,
+            buf: iter::repeat(0).take(bsize).collect(),
+            dlt: iter::repeat(0).take(dsize).collect(),
```

Comment:
// simply use vec! here
buf: vec![0; bsize],
dlt: vec![0; dsize],

Reply:
Indeed, `vec![0; bsize];` is not only more concise but also faster. In fact, it will simply request already zeroed memory from the OS if possible, so zeroing the memory is essentially free if done once.